### PR TITLE
[REVIEW] Add CHANGELOG.md

### PR DESCRIPTION
**Pull request by harrism**
From now on every PR author should add a human-readable explanation of the PR at the appropriate location in this file. This will be a pre-condition for merging the PR. This PR does so.

### CODE REVIEW

**Comment by kkraus14** [comment]
Can we start the changelog with 0.3.0? Not sure what value providing the changelog for 0.2.x and 0.1.x adds.

**Comment by harrism** [comment]
I thought the same thing. That's fine with me. I could use some help filling out 0.3.0.

**Comment by kkraus14** [comment]
Yes, cc @randerzander and I should be able to jump in and help with this.

**Comment by harrism** [comment]
I went through the PRs that went into 0.3.0 and added them. I also added the changelog to 0.4.0. Any further additions to 0.4.0 (and the date) should be added in another PR.

I think this PR is ready for review.

**Comment by kkraus14** [comment]
Will let @mike-wendt merge pending his review.

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+# cuDF 0.3.0 (23 Nov 2018)
+
+## New Features
+
+ - PR #336 CSV Reader string support
+
+## Improvements
+ 
+ - PR #354 source code refactored for better organization. CMake build system overhaul.
```
Comment (dantegd) [inline CHANGELOG.md]: Perhaps mention `beginning of Cython bindings transition` that was part of the refactor PR.

Comment (harrism) [reply CHANGELOG.md]: Done.

**Comment by harrism** [comment]
rerun tests

**Comment by randerzander** [comment]
rerun tests

**Pull request merged**